platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/IsolatedExtensionsContainer.kt

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.cc.impl
+
+import org.gradle.api.Action
+import org.gradle.api.InvalidUserCodeException
+import org.gradle.api.internal.plugins.ExtensionContainerInternal
+import org.gradle.api.internal.properties.GradleProperties
+import org.gradle.api.plugins.ExtensionsSchema
+import org.gradle.api.plugins.ExtraPropertiesExtension
+import org.gradle.api.reflect.TypeOf
+import org.gradle.api.reflect.TypeOf.typeOf
+import org.gradle.internal.extensibility.DefaultExtensionsSchema
+import org.gradle.internal.extensibility.DefaultExtraPropertiesExtension
+import org.gradle.internal.extensions.stdlib.uncheckedCast
+import org.gradle.internal.isolate.graph.IsolatedActionDeserializer
+import org.gradle.internal.isolate.graph.IsolatedActionSerializer
+import org.gradle.internal.isolate.graph.SerializedIsolatedActionGraph
+
+internal class IsolatedExtensionsContainer(
+    private val serializer: IsolatedActionSerializer,
+    private val deserializer: IsolatedActionDeserializer,
+    private val delegate: ExtensionContainerInternal,
+    private val gradleProperties : GradleProperties
+) : ExtensionContainerInternal {
+
+    private val serializedExtensionsByType = mutableMapOf<TypeOf<*>, SerializedIsolatedActionGraph<Any>>()
+    private val serializedExtensionsByName = mutableMapOf<String, SerializedIsolatedActionGraph<Any>>()
```

Comment:
The name `SerializedIsolatedActionGraph` doesn't really make much sense anymore now that we're serializing more than just actions.